sphinx/ext/autodoc/__init__.py

```diff
@@ -1504,11 +1477,11 @@ def add_content(self, more_content, no_docstring=False):
         AttributeDocumenter.add_content(self, more_content, no_docstring=True)
 
 
-class AutoDirective(Directive):
+class AutoDirective(object):
```

Comment:
You should probably take the opportunity to rename this now that it's no longer a directive. The name is very confusing.

Reply:
Indeed. but many third party codes touches `AutoDirective._registry` directly...
https://github.com/search?q=AutoDirective._registry&type=Code&utf8=%E2%9C%93

So we have to deprecate this step by step. Now we have `sphinx.registry` as a components registry for Sphinx. So this will be replaced and removed in future.

Reply:
I agreed to keeping name of AutoDirective. However I think it would be nice If we keep compatibility while changing to the proper name like the following code:
```
class AutodocRegistry(object):
    ...

AutoDirective = AutodocRegistry  # for the backward compatibility.
```

Reply:
Okay, I just renamed.